File: backend/product_service/src/main/java/ucv/app_inventory/application/services/CategoryService.java
package ucv.app_inventory.application.services;

import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import ucv.app_inventory.domain.entities.Category;

import java.util.List;

public interface CategoryService {
    List<Category> listCategories(int page, int size);
    Category createCategory(Category category);
    Category updateCategory(Category category);
    Category findCategoryById(Long id);
    void deleteCategory(Category category);
    Page<Category> findByName(String name, Pageable pageable);
    Page<Category> findByStatus(Category.Status status, int page, int size);
    Page<Category> findByNameAndStatus(String name, Category.Status status, int page, int size);
}

File: frontend/src/services/products/CategoryService.js
// services/products/CategoryService.js
import config from '../../config';
import ApiCategory from './apiCategory';

const API_CATEGORIES = config.API_CATEGORIES;

const CategoryService = {
    listCategories: () =>
        ApiCategory(`${API_CATEGORIES}/list`),

    saveCategory: (category) =>
        ApiCategory(`${API_CATEGORIES}/save`, 'POST', category),

    updateCategory: (id, updatedCategory) =>
        ApiCategory(`${API_CATEGORIES}/update/${id}`, 'PUT', updatedCategory),

    deleteCategory: (id) =>
        ApiCategory(`${API_CATEGORIES}/delete/${id}`, 'DELETE'),

    findCategoryById: (id) =>
        ApiCategory(`${API_CATEGORIES}/${id}`),

    findByName: (name, page = 0, size = 15) =>
        ApiCategory(`${API_CATEGORIES}/findByName?name=${name}&page=${page}&size=${size}`),

    findByStatus: (status, page = 0, size = 15) =>
        ApiCategory(`${API_CATEGORIES}/findByStatus?status=${status}&page=${page}&size=${size}`),

};

export default CategoryService;
